### ISSUE TITLE
DateField and DateTimeField

### ISSUE THREAD

**Issue by tymarats** [codaxy/cxjs#442]
It would be very useful if `DateField` and `DateTimeField` would have an option (`includeOffset` maybe, or something similar) to be backed up by a model with offset included (as per ISO 8601), e.g.: `2018-06-07T00:00:00+01`. 

Right now, backing model is always a Zulu-time string, which is not optimal for all situations (actually, for most situations if we're dealing with dates only).

**Comment by mstijak** [comment]
I agree. We should add a new flag which controls how are dates written to the store. Which name should we use? `encoding`?

I see three good candidates:

1) "iso":  date => date.toISODateString()
2) "timestamp": date => date.valueOf()
3) "iso8601+offset": date => encodeDateWithTimezoneOffset(date)

Please suggest alternative names. It should be possible to pass a function too.

```
<Calendar 
  value-bind="date"
  encoding={encodeDateWithTimezone} 
/>
```

**Comment by tymarats** [comment]
Possibility of passing in a function is a great idea, so that we can provide any format that's `Date.parse`-able; for example, ISO without the offset suffix, or even the date part only. Then we can POST these values to even wider array of APIs directly (i.e. without any further transformations of the model values beforehand).

As for the flag name, `encoding` seems fine and succint enough.

**Comment by mstijak** [comment]
After more consideration, I would suggest a slightly different approach.

If we introduce named encodings (e.g. "iso", "timestamp", "iso+timezone" or whatever) that means that we need to preregister all of them. That's bad as all applications will bundle the function `encodeDateWithTimezoneOffset` regardless if they use it or not. 

It would be better that `encoding` accepts only function arguments. Also it should be possible to set it application wide. The `Culture` class could be extended with `setDefaultDateEncoding` and `getDefaultDateEncoding`.

You would then write `Culture.setDefaultDateEncoding(encodeDateWithTimezoneOffset)` once to change the behavior application-wide.

**Comment by tymarats** [comment]
Sounds reasonable.

I can see how `Culture` is a convenient place to put this global encoding handling, even though semantically this doesn't have much to do with the culture. I'm guessing it's not that easy to make it a "global" function, like when you're enabling optional features (like `enableMsgBoxAlerts`, for example)? In any case, with appropriate documentation, any solution is fine.

**Comment by mstijak** [comment]
Implemented in cx@16.6.5!
